### PR TITLE
update flow docs

### PR DESCRIPTION
**Pull request by jasonLaster**
Here are some updates based on my discussion with @rpl and stealing from their docs: https://github.com/mozilla/web-ext/blob/master/CONTRIBUTING.md#check-for-flow-errors

CC @rpl, @clarkbw, @thejameskyle

### CODE REVIEW

**Comment by clarkbw** [comment]
Looks good so far, I'll take a closer look later

File: docs/flow.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
+## Flow
+
+- [Adding flow to a file](#adding-flow-to-a-file)
+- [Running flow](#running-flow)
+- [Missing Annotation](#missing-annotation)
+- [Where are types defined?](#where-are-types-defined?)
```
Comment (clarkbw) [inline docs/flow.md]: I think you need to drop the `?` here `#where-are-types-defined`

File: docs/flow.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+- [Checking flow coverage](#checking-flow-coverage)
+- [Flow Conventions][#flow-conventions]
```
Comment (clarkbw) [inline docs/flow.md]: Typo, needs `( )` here: `[Flow Conventions](#flow-conventions)`

File: docs/flow.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+The debugger uses Facebook's [flow](https://flowtype.org/) type checker.
+
+Rationale:
+* *code clarity* - helps team members understand the code
+* *refactoring* - guarantees functions integrate well
+* *code reviews* - adds a static check like linting
+
+### Running flow
+```
+> yarn run flow
```
Comment (clarkbw) [inline docs/flow.md]: We're a little inconsistent in our docs but I don't think we need the leading `>` for these commands.

Comment (jasonLaster) [inline docs/flow.md]: ahh, yeah agreed

File: docs/flow.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+```
+
+Go checkout the [Flow related issues][flow-issues]
+
+### Adding flow to a file
+
+Add `// @flow` to the top of the file.
+
+```patch
```
Comment (clarkbw) [inline docs/flow.md]: `patch` doesn't seem to work with highlighting anymore, but `diff` does.

Comment (jasonLaster) [inline docs/flow.md]: oh yeah!

**Pull request merged**